farmdata2_modules/fd2_tabs/resources/DateRangeSelectionComponent.spec.comp.js

```diff
@@ -1,73 +1,141 @@
+                .children()
+                    .type('2056-01-01')
+                    .blur()
+                    .should('have.value', '2021-12-01')
+        })
 
-    it('fail to change end date to before start date', () => {
-        cy.get('[data-cy=end-date-select]')
-            .children()
-                .type('2000-01-01')
-                .blur()
-                .should('have.value', '2021-01-01')
-    })
+        it('fail to change end date to before start date', () => {
+            cy.get('[data-cy=end-date-select]')
+                .children()
+                    .type('2000-01-01')
+                    .blur()
+                    .should('have.value', '2021-01-01')
+        })
+
+        it('emits the new selected start date', () => {
+            const spy = cy.spy()
+            Cypress.vue.$on('start-date-changed', spy)
+            cy.get('[data-cy=start-date-select]')
+                .children()
+                    .type('2021-05-01')
+                    .blur()
+                    .then(() => {
+                        expect(spy).to.be.called
+                    })
+        })
+
+        it('emits on click', () => {
+            const spy = cy.spy()
+            Cypress.vue.$on('click', spy)
+            cy.get('[data-cy=start-date-select]')
+                .children().click()
+                    .then(() => {
+                        expect(spy).to.be.called
+                    })
+        })
 
-    it('emits the new selected start date', () => {
-        const spy = cy.spy()
-        Cypress.vue.$on('start-date-changed', spy)
-        cy.get('[data-cy=start-date-select]')
-            .children()
-                .type('2021-05-01')
-                .blur()
-                .then(() => {
-                    expect(spy).to.be.called
-                 })
+        it('emits the new selected end date', () => {
+            const spy = cy.spy()
+            Cypress.vue.$on('end-date-changed', spy)
+            cy.get('[data-cy=end-date-select]')
+                .children()
+                    .type('2021-07-01')
+                    .blur()
+                    .then(() => {
+                        expect(spy).to.be.calledWith('2021-07-01')
+                    })
+        })
     })
+    context('watch prop change test', () => {
+        let comp;
+        beforeEach(() => {  
+            // Use shallowMount here so we can use setProp
+            comp = shallowMount(DateRangeSelectionComponent, {
+                propsData: {
+                    setStartDate: "2021-01-01",
+                    setEndDate: "2021-12-31"
+                }   
+            })
+        })
+
+        it ('change setStartDate prop to 2021-08-09', () => {
+            expect(comp.vm.earliestEndDate).to.equal('2021-01-01')
+            cy.wrap(comp.setProps({ setStartDate: '2021-08-09' }))
+            .then(() => {
+                expect(comp.vm.earliestEndDate).to.equal('2021-08-09')
+            })
+        })
+
```

Comment:
There should also be a separate test here that checks that changes to the end date work.